### PR TITLE
payer plan period criteria support for circe-be

### PR DESCRIPTION
**Pull request by raomane**
Related to https://github.com/OHDSI/CommonDataModel/issues/120
If/Once the proposal is accepted
@chrisknoll 
https://github.com/OHDSI/WebAPI/issues/217

### CODE REVIEW

File: src/main/java/org/ohdsi/circe/cohortdefinition/PayerPlanPeriod.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	public DateRange periodEndDate;
+	
+	@JsonProperty("UserDefinedPeriod")
+	public Period userDefinedPeriod;
+	
+	@JsonProperty("PeriodLength")
+	public NumericRange periodLength;
+	
+	@JsonProperty("AgeAtStart")
+	public NumericRange ageAtstart;
```
Comment (mpozhidaeva) [inline src/main/java/org/ohdsi/circe/cohortdefinition/PayerPlanPeriod.java]: @raomane, @gowthamrao - hello, do you mean "ageAtStart"?

File: src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java [hunk 2]
```diff
@@ -1570,7 +1571,147 @@ public String getCriteriaSql(ObservationPeriod criteria)
...
+	
+	if (criteria.ageAtStart != null || criteria.ageAtEnd != null || (criteria.gender != null && criteria.gender.length > 0))
+	  joinClauses.add("JOIN @cdm_database_schema.PERSON P on C.person_id = P.person_id");
+	
+	query = StringUtils.replace(query, "@joinClause", StringUtils.join(joinClauses, "\n"));
+	
+	ArrayList<String> whereClauses = new ArrayList<>();
+	
+	//first
+	if (criteria.first != null && criteria.first == true)
```
Comment (mpozhidaeva) [inline src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java]: I think `... && criteria.first` is enough

File: src/main/java/org/ohdsi/circe/cohortdefinition/PayerPlanPeriod.java (hunk 1 above)
Comment (gowthamrao) [reply src/main/java/org/ohdsi/circe/cohortdefinition/PayerPlanPeriod.java]: agree

File: src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java (hunk 2 above)
Comment (gowthamrao) [reply src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java]: i think this is inline with other patterns. 
e.g.
https://github.com/OHDSI/circe-be/blob/6b602fd0b405e0cfe0846e9189b613888326ab72/src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java#L633
https://github.com/OHDSI/circe-be/blob/6b602fd0b405e0cfe0846e9189b613888326ab72/src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java#L717
https://github.com/OHDSI/circe-be/blob/6b602fd0b405e0cfe0846e9189b613888326ab72/src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java#L879

etc

**Pull request merged**